bgpd/bgpd.h

```diff
@@ -774,8 +779,17 @@ typedef enum {
 #define BGP_MARKER_SIZE		                16
 #define BGP_HEADER_SIZE		                19
 #define BGP_MAX_PACKET_SIZE                   4096
+#define BGP_MAX_EXT_MESSAGE_PACKET_SIZE      65535
 #define BGP_MAX_PACKET_SIZE_OVERFLOW          1024
 
+/* Use bigger buffer size if BGP Extended message capability is set
+ * locally and remotely.
+ */
+#define BGP_PACKET_SIZE(peer)                                                  \
+	((CHECK_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_ADV)                      \
+	  && CHECK_FLAG(peer->cap, PEER_CAP_EXT_MESSAGE_RCV))                  \
+		 ? BGP_MAX_EXT_MESSAGE_PACKET_SIZE                             \
+		 : BGP_MAX_PACKET_SIZE)
```

Comment:
This macro is called a lot during processing.  I suggest you cache the value in the peer structure so it can be directly returned.